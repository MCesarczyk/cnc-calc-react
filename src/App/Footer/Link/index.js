import React from "react";
import Link from "../../Link";

const FooterLink = ({ adress, text, prefix }) => (
  <>
    {prefix}
    {' '}
    <Link
      adress={adress}
      text={text}
    />
  </>
);

export default FooterLink;