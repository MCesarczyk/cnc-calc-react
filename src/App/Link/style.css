.normal {
  color: inherit;
}

.hovered {
  color: crimson;
  border-bottom: 1px solid currentColor;
  padding-bottom: 4px;
}