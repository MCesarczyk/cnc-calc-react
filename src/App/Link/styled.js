import styled, { css } from "styled-components";

export const StyledLink = styled.a.attrs(props => ({
  hovered: props.status === 'hovered',
  normal: props.status === 'normal'
}))`
  color: inherit;

  ${props => props.hovered && css`
    color: crimson;
    border-bottom: 1px solid currentColor;
    padding-bottom: 4px;
  `}
`