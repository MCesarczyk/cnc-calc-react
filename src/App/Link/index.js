import React, { useState } from "react";
import "./style.css";

const STATUS = {
  HOVERED: 'hovered',
  NORMAL: 'normal',
};

const Link = ({ adress, text }) => {
  const [status, setStatus] = useState(STATUS.NORMAL);

  const onMouseEnter = () => {
    setStatus(STATUS.HOVERED);
  };

  const onMouseLeave = () => {
    setStatus(STATUS.NORMAL);
  };

  return (
    <a
      className={status}
      href={adress || '#'}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      {text}
    </a>
  )
};

export default Link;