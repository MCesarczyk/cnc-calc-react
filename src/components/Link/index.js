import { StyledLink } from "./styled";

const Link = ({ adress, text }) => (
  <StyledLink
    href={adress || '#'}
  >
    {text}
  </StyledLink>
);

export default Link;