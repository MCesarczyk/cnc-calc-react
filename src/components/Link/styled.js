import styled from "styled-components";

export const StyledLink = styled.a`
  color: inherit;

  &:hover {
    color: crimson;
    border-bottom: 1px solid currentColor;
    padding-bottom: 4px;
  }
`