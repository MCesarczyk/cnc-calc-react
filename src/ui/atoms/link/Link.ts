import styled from 'styled-components';

interface LinkProps {
  href: string;
}

export const Link = styled.a<LinkProps>`
  color: inherit;
  transition: 0.5s ease-out;

  &:hover {
    color: crimson;
  }
`;
