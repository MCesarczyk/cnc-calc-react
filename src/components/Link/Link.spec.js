import React from "react";
import renderer from "react-test-renderer";
import Link from ".";
import "jest-styled-components";

test("Link with states rendering properly", () => {
  const component = renderer.create(
    <Link />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});