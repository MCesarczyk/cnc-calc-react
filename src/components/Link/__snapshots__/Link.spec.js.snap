// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link with states rendering properly 1`] = `
.c0 {
  color: inherit;
}

.c0:hover {
  color: crimson;
  border-bottom: 1px solid currentColor;
  padding-bottom: 4px;
}

<a
  className="c0"
  href="#"
/>
`;
